### PR TITLE
Fix cmdline for stm32 and add entry in readme

### PR DESCRIPTION
**Pull request by akhilveeraghanta**
### Description

The executable requires cwd to be the folder w/ the xml and jar files, so running `STM32CubeMX` from the cmd line now cd's and then executes there, and when finished, cds back to the previous dir. 

Also added an entry in `getting-started.md` w/ the command to regen cube code.

### Testing Done

Ran the installer and made sure I can run cube from command line
### Length Justification
N/A

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: environment_setup/install_cubemx.sh [hunk 1]
```diff
@@ -39,12 +39,22 @@ else
         </AutomatedInstallation>"
     
     echo -n $AUTO_INSTALL_XML > auto-install.xml
+
+    # create script to cd and run STM32CubeMX
+    STM32_SCRIPT="cd /opt/STM32CubeMX_$CUBE_VERSION && ./STM32CubeMX && cd -"
```
Comment (jonathanlew) [inline environment_setup/install_cubemx.sh]: Consider using pushd and popd

**Review by jonathanlew — COMMENTED** [review]
:+1:

File: environment_setup/install_cubemx.sh (hunk 1 above)
Comment (garethellis0) [reply environment_setup/install_cubemx.sh]: :+1: for this, much less error-prone

File: docs/getting-started.md [hunk 2]
```diff
@@ -153,3 +153,9 @@ Debugging in CLion is as simple as running the above instructions for building C
...
 3. Assuming you choose 0 from the list in step (2), run `bazel run --cpu=stm32h7 --compilation_mode=dbg //firmware_new/tools:debug_firmware_on_arm_board 0`. This will load the `.elf` file associated with (0) to the the nucleo and put you into a gdb prompt.
 4. At this point you should be in a gdb window. Take a look at [this tutorial](https://www.cprogramming.com/gdb.html) for some basics.
+
+## Working with CubeMX to regenerate code
+1. Make sure you've followed [Installing Firmware Dependencies](#installing-firmware-dependencies)
+2. To regenerate code from the `.ioc` file, run `bazel run //firmware_new/tools:cubemx_regen firmware_new/boards/frankie_v1`. The directory that is passed in as an argument must only contain 1 ioc file, which will be used to generate code into the same directory.
```
Comment (garethellis0) [inline docs/getting-started.md]:
``` suggestion
2. To regenerate code from the `.ioc` file, run `bazel run //firmware_new/tools:cubemx_regen path/to/directory/with/.ioc`. The directory that is passed in as an argument must only contain 1 ioc file, which will be used to generate code into the same directory.
```

**Review by garethellis0 — CHANGES_REQUESTED** [review]

File: environment_setup/install_cubemx.sh (hunk 1 above)
Comment (akhilveeraghanta) [reply environment_setup/install_cubemx.sh]: not opposed to changing this, just wondering what makes pushd/popd less error prone?

Comment (garethellis0) [reply environment_setup/install_cubemx.sh]: Perhaps "much" was a bit of a strong word, but IMO writing:
```
pushd /tmp/stm32cubemx

do something 
do something else

popd
```
provides a much clearer indication that you're moving into a directory, then back to the directory you started in. Although TIL what `cd -` does, that's basically the same. Don't feel super strongly here either way, leaving this up to you / Jon.

Comment (akhilveeraghanta) [reply environment_setup/install_cubemx.sh]: yeah its definitely more explicit, fixed!

**Review by garethellis0 — DISMISSED** [review]
LGTM :+1: :tada:

**Comment by akhilveeraghanta** [comment]
Sorry I missed this earlier, the `cube_regen` script doesn't pass the arguments properly to cube because now we have a script at` /usr/local/bin/STM...` vs a jar, `c2b70a6` is fixing that path to be the jar rather than the script

**Review by garethellis0 — DISMISSED** [review]

**Review by jonathanlew — DISMISSED** [review]
:+1:

**Review by MathewMacDougall — DISMISSED** [review]

**Comment by akhilveeraghanta** [comment]
Caught one more issue w/ the dir, my tmp directory expired and the script broke, when switching to pushd I used the wrong dir, this should hopefully be the last time, sorry about the spam

File: environment_setup/install_cubemx.sh [hunk 3]
```diff
@@ -39,12 +39,22 @@ else
         </AutomatedInstallation>"
     
     echo -n $AUTO_INSTALL_XML > auto-install.xml
+
+    # create script to cd and run STM32CubeMX
+    STM32_SCRIPT="pushd . && cd /opt/STM32CubeMX_$CUBE_VERSION && ./STM32CubeMX && popd"
```
Comment (jonathanlew) [inline environment_setup/install_cubemx.sh]:
```suggestion
    STM32_SCRIPT="pushd /opt/STM32CubeMX_$CUBE_VERSION && ./STM32CubeMX && popd"
```

**Review by jonathanlew — CHANGES_REQUESTED** [review]
Is this what you meant?

File: environment_setup/install_cubemx.sh (hunk 3 above)
Comment (akhilveeraghanta) [reply environment_setup/install_cubemx.sh]: yeah lol, took 3 tries to get it right

**Review by garethellis0 — APPROVED** [review]

**Review by MathewMacDougall — APPROVED** [review]

**Review by jonathanlew — APPROVED** [review]

**Pull request merged**